pkg/capi/gcp.go

```diff
@@ -0,0 +1,308 @@
+		Entry("SecureBoot and IntegrityMonitoring enabled", gcpv1.SecureBootPolicyEnabled, gcpv1.VirtualizedTrustedPlatformModulePolicyDisabled, gcpv1.IntegrityMonitoringPolicyEnabled),
+		Entry("all Shielded VM options disabled", gcpv1.SecureBootPolicyDisabled, gcpv1.VirtualizedTrustedPlatformModulePolicyDisabled, gcpv1.IntegrityMonitoringPolicyDisabled),
+		*/
+	)
+	// Define constants as variable for this case to pass values properly
+	confidentialComputeEnabled := gcpv1.ConfidentialComputePolicyEnabled
+	confidentialComputeDisabled := gcpv1.ConfidentialComputePolicyDisabled
+	DescribeTable("should configure Confidential VM correctly", framework.LabelCAPI, framework.LabelDisruptive,
+		func(confidentialCompute *gcpv1.ConfidentialComputePolicy) {
+			mapiProviderSpec := getGCPMAPIProviderSpec(cl)
+			Expect(mapiProviderSpec).ToNot(BeNil())
+			gcpMachineTemplate = createGCPMachineTemplate(mapiProviderSpec)
+			gcpMachineTemplate.Spec.Template.Spec.ConfidentialCompute = confidentialCompute
+			gcpMachineTemplate.Spec.Template.Spec.InstanceType = "n2d-standard-4"
+
+			if *confidentialCompute == gcpv1.ConfidentialComputePolicyEnabled {
+				mapiProviderSpec.OnHostMaintenance = OnHostMaintenanceTerminate
+				gcpMachineTemplate.Spec.Template.Spec.OnHostMaintenance = (*gcpv1.HostMaintenancePolicy)(&mapiProviderSpec.OnHostMaintenance)
+			} else {
+				mapiProviderSpec.OnHostMaintenance = "Migrate"
+				gcpMachineTemplate.Spec.Template.Spec.OnHostMaintenance = (*gcpv1.HostMaintenancePolicy)(&mapiProviderSpec.OnHostMaintenance)
+			}
+
+			if err := cl.Create(ctx, gcpMachineTemplate); err != nil && !apierrors.IsAlreadyExists(err) {
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			By("Creating a MachineSet for Confidential VM")
+			machineSet, err = framework.CreateCAPIMachineSet(ctx, cl, framework.NewCAPIMachineSetParams(
+				"gcp-machineset-confidential-74703",
+				clusterName,
+				mapiProviderSpec.Zone,
+				1,
+				corev1.ObjectReference{
+					Kind:       "GCPMachineTemplate",
+					APIVersion: infraAPIVersion,
+					Name:       gcpMachineTemplateName,
+				},
+			))
+			Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI MachineSet with Confidential VM configuration")
+
+			framework.WaitForCAPIMachinesRunning(ctx, cl, machineSet.Name)
+
+			By("Verifying the Confidential VM configuration on the created GCP MachineTemplate")
+			createdTemplate := &gcpv1.GCPMachineTemplate{}
+			Expect(cl.Get(framework.GetContext(), client.ObjectKey{
+				Namespace: framework.ClusterAPINamespace,
+				Name:      gcpMachineTemplateName,
+			}, createdTemplate)).To(Succeed())
+			var confidentialComputevalue = *createdTemplate.Spec.Template.Spec.ConfidentialCompute
+			Expect(fmt.Sprintf("%v", confidentialComputevalue)).To(Equal(fmt.Sprintf("%v", *confidentialCompute)))
+		},
+		Entry("Confidential Compute enabled", &confidentialComputeEnabled),
```

Comment:
You could use `ptr.To` rather than taking the pointer like this

From `k8s.io/utils/ptr`

We use it a lot in other tests

Reply:
I was having some trouble , with validation if I was using ptr.To so , I removed pointer value from compare to a simple string , will include that in next commit.

Reply:
value of `*confidentialCompute` is `Enabled or Disabled` I thought return value of ptr.To was boolean `true/false` whereas it can be pointer to any literal ,  will modify the code as below - 
```
			expectedConfidentialComputeValue := ptr.To(
				map[bool]string{
					true:  "Enabled",
					false: "Disabled",
				}[*confidentialCompute == gcpv1.ConfidentialComputePolicyEnabled],
			)

			Expect(fmt.Sprintf("%v", *expectedConfidentialComputeValue)).To(Equal(fmt.Sprintf("%v", *confidentialCompute)))
```